framework/src/modules/chain/bft/bft.js

```diff
@@ -84,12 +85,62 @@ class BFT extends EventEmitter {
 			this.constants.startingHeight,
 		);
 
-		await this.loadBlocks({
+		await this.loadBlocksFromStorage({
 			fromHeight: loadFromHeight,
 			tillHeight: lastBlockHeight,
 		});
 	}
 
+	/**
+	 * When blocks deleted send those to BFT to update BFT state
+	 *
+	 * @param {Array.<Object>} blocks - List of all blocks
+	 * @return {Promise<void>}
+	 */
+	async onDeleteBlocks(blocks) {
+		assert(blocks, 'Must provide blocks which are deleted');
+		assert(Array.isArray(blocks), 'Must provide list of blocks');
+
+		// We need only height to delete the blocks
+		// But for future extension we accept full blocks in BFT
+		// We may need to utilize some other attributes for internal processing
+		const blockHeights = blocks.map(({ height }) => height);
+
+		assert(
+			!blockHeights.some(h => h <= this.finalityManager.finalizedHeight),
+			'Can not delete block below or same as finalized height',
+		);
+
+		const removeFromHeight = Math.min(...blockHeights);
+
+		this.finalityManager.removeBlockHeaders({
+			aboveHeight: removeFromHeight - 1,
```

Comment:
Can you clarify with a comment why `removeFromHeight - 1` is applied here? Thanks! :)

Reply:
Because of `removeBlockHeaders` have `aboveHeight` param. Which removes block headers above a point. So if you want to delete the current one, you have to do a step back.

Reply:
Would it make sense to add this info as a `//` comment above this line of code? Thanks!